app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,103 @@
+class TweetsController < ApplicationController
+	get '/tweets' do  
+		if Helpers.logged_in?(session)
+			@user=Helpers.current_user(session)
+			erb :'tweets/index'
+		else
+			redirect '/login'
+		end
+	end
+
+	get '/tweets/new' do  
+		if Helpers.logged_in?(session)
+			@user=Helpers.current_user(session)
+			erb :'tweets/new'
+		else
+			redirect '/login'
+		end
+		
+	end
+
+	get '/tweets/:slug' do  
+		if Helpers.logged_in?(session)
+			@current_user=Helpers.current_user(session)
+			@tweet = Tweet.find(params[:slug].to_i)
+			erb :'tweets/view_tweet'
+		else
+			redirect '/login'
+		end
+		
+	end
+
+	post '/tweets/:slug/delete' do  
+		#binding.pry
+		@tweet = Tweet.find(params[:slug].to_i)
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(:id => params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the users tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully simply returning nil if the user is trying to delete a tweet that they do not have access to.